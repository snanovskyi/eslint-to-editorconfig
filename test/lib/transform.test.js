const transform = require('../../lib/transform');

describe('transform', () => {
  it('should return EditorConfig rules', () => {
    const eslintRules = {
      indent: 'error',
      'no-trailing-spaces': 'error',
      'eol-last': 'error',
      'max-len': 'error',
    };
    const result = transform(eslintRules);
    expect(result).toMatchSnapshot();
  });

  it('should return rules that are defined', () => {
    const eslintRules = {
      indent: 'error',
      'max-len': 'error',
    };
    const result = transform(eslintRules);
    expect(result).toMatchSnapshot();
  });

  it("should throw 'TypeError' if ESLint rules is not an object", () => {
    const error1 = () => transform(undefined);
    const error2 = () => transform(123);
    const error3 = () => transform('123');
    const error4 = () => transform(() => {});
    const error5 = () => transform(Symbol('123'));
    expect(error1).toThrow(TypeError);
    expect(error2).toThrow(TypeError);
    expect(error3).toThrow(TypeError);
    expect(error4).toThrow(TypeError);
    expect(error5).toThrow(TypeError);
  });
});
