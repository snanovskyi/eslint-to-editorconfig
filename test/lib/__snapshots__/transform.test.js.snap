// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform should return EditorConfig rules 1`] = `
Object {
  "indent_size": 4,
  "indent_style": "space",
  "insert_final_newline": true,
  "max_line_length": 80,
  "trim_trailing_whitespace": true,
}
`;

exports[`transform should return rules that are defined 1`] = `
Object {
  "indent_size": 4,
  "indent_style": "space",
  "max_line_length": 80,
}
`;
